packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -676,128 +653,146 @@ describe("SharedTree", () => {
 		revertible.revert();
 
 		validateTreeContent(summarizingTree.checkout, {
-			schema: stringSequenceRootSchema,
+			schema: toFlexSchema(stringArray),
 			initialTree: ["a", "b", "c"],
 		});
 
 		await provider.ensureSynchronized();
 
 		validateTreeContent(loadingTree.checkout, {
-			schema: stringSequenceRootSchema,
+			schema: toFlexSchema(stringArray),
 			initialTree: ["a", "b", "c"],
 		});
 		unsubscribe();
 	});
 
 	it("can summarize local edits in the attach summary", async () => {
 		const onCreate = (tree: SharedTree) => {
-			const view = schematizeFlexTree(tree, emptyStringSequenceConfig);
-			view.flexTree.insertAtStart(["A"]);
-			view.flexTree.insertAtEnd(["C"]);
-			assert.deepEqual([...view.flexTree], ["A", "C"]);
-			view[disposeSymbol]();
+			const view = tree.viewWith({ schema: stringArray, enableSchemaValidation });
+			view.initialize([]);
+			view.root.insertAtStart("A");
+			view.root.insertAtEnd("C");
+			assert.deepEqual([...view.root], ["A", "C"]);
+			view.dispose();
 		};
 		const provider = await TestTreeProvider.create(
 			1,
 			SummarizeType.onDemand,
 			new SharedTreeTestFactory(onCreate),
 		);
-		const tree1 = assertSchema(provider.trees[0], stringSequenceRootSchema);
-		assert.deepEqual([...tree1.flexTree], ["A", "C"]);
+		const tree1 = provider.trees[0];
+		const view1 = tree1.viewWith({ schema: stringArray, enableSchemaValidation });
+		assert.deepEqual([...view1.root], ["A", "C"]);
 		await provider.ensureSynchronized();
-		const tree2 = assertSchema(await provider.createTree(), stringSequenceRootSchema);
+		const tree2 = await provider.createTree();
+		const view2 = tree2.viewWith({ schema: stringArray, enableSchemaValidation });
+
 		// Check that the joining tree was initialized with data from the attach summary
-		assert.deepEqual([...tree2.flexTree], ["A", "C"]);
+		assert.deepEqual([...view2.root], ["A", "C"]);
 
 		// Check that further edits are interpreted properly
-		tree1.flexTree.insertAt(1, "B");
+		view1.root.insertAt(1, "B");
 		await provider.ensureSynchronized();
-		assert.deepEqual([...tree1.flexTree], ["A", "B", "C"]);
-		assert.deepEqual([...tree2.flexTree], ["A", "B", "C"]);
+		assert.deepEqual([...view1.root], ["A", "B", "C"]);
+		assert.deepEqual([...view2.root], ["A", "B", "C"]);
 	});
 
 	it("can tolerate local edits submitted as part of a transaction in the attach summary", async () => {
 		const onCreate = (tree: SharedTree) => {
 			// Schematize uses a transaction as well
-			const view = schematizeFlexTree(tree, emptyStringSequenceConfig);
+			const view = tree.viewWith({ schema: stringArray, enableSchemaValidation });
+			view.initialize([]);
 			view.checkout.transaction.start();
```

Comment:
we do a lot of starting transactions via the checkout in this file. could we use the customer-facing API instead?

Reply:
Yea good idea, I'll update where applicable.